spec/jobs/inherit_permissions_job_spec.rb

```diff
@@ -2,92 +2,199 @@
+          Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+            expect(file.edit_users).to match_array [user.to_s, user2.to_s]
+          end
+
+          described_class.perform_now(resource, use_valkyrie: true)
+
+          # files have single edit user from parent resource
+          Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+            expect(file.edit_users).to match_array [user.to_s]
+          end
+        end
+      end
+    end
+
+    context "when read people change" do
+      it 'copies permissions to its contained files' do
+        resource.permission_manager.acl.grant(:read).to(user2).save
+        expect(resource.edit_users).to match_array [user.to_s]
+
+        # files have no read users to begin with
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+          expect(file.read_users.to_a).to be_empty
+        end
+
+        described_class.perform_now(resource, use_valkyrie: true)
+
+        # files have the extra user as the read user
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+          expect(file.read_users.to_a).to match_array [user2.to_s]
+        end
+      end
+    end
+
+    context "when read groups change" do
+      let(:group) { Hyrax::Group.new('my_read_group') }
+
+      it 'copies permissions to its contained files' do
+        resource.permission_manager.acl.grant(:read).to(group).save
+
+        # files have the depositor as the edit user to begin with
+        expect(resource.read_groups).to match_array ["my_read_group"]
+
+        described_class.perform_now(resource, use_valkyrie: true)
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+          expect(file.edit_users).to match_array [user.to_s]
+          expect(file.read_groups).to match_array ["my_read_group"]
+        end
+      end
+    end
+
+    context "when edit groups change" do
+      let(:group) { Hyrax::Group.new('my_edit_group') }
 
-    it 'copies permissions to its contained files' do
-      # files have the depositor as the edit user to begin with
-      expect(work.file_sets.first.read_groups).to eq []
+      it 'copies permissions to its contained files' do
+        resource.permission_manager.acl.grant(:edit).to(group).save
+
+        # files have the depositor as the edit user to begin with
+        expect(resource.edit_groups).to match_array ["my_edit_group"]
 
-      described_class.perform_now(work)
-      work.reload.file_sets.each do |file|
-        expect(file.edit_groups).to match_array ["my_edit_group"]
-        expect(file.edit_users).to match_array [user.to_s]
+        described_class.perform_now(resource, use_valkyrie: true)
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
```

Comment:
In reading this spec, it would be possible that we have an empty array, which would mean the spec is not testing anything.

It may be helpful to capture the `Hyrax.query_service.custom_queries.find_child_filesets(resource: resource)` as a variable and say that you expect it to have a specific count/size.